apis/config/v1alpha1/clusterconfig_types.go

```diff
@@ -39,6 +39,9 @@ type ClusterConfigSpec struct {
 	AgentConfig AgentConfig `json:"agentConfig"`
 }
 
+// +kubebuilder:validation:Pattern="^([0-9]{1,3}.){3}[0-9]{1,3}(/([0-9]|[1-2][0-9]|3[0-2]))"
```

Comment:
I'm not a pro in regular expressions, but the pattern specified here allows CIDR written as `192.168.1.0/24ast`. So without a end anchor we can put whatever we want after the last group. We should say that the last group must exist one and only once and to be also the end of the string. something like this `^([0-9]{1,3}.){3}[0-9]{1,3}(/([0-9]|[1-2][0-9]|3[0-2])){1}$`.